connection.go

```diff
@@ -152,20 +152,19 @@ func (ct *ConnectionTransportTLS) Dial(ctx context.Context) (
 	var conn net.Conn
 	err := runUnlessCanceled(ctx, func() error {
 		// load CA certificate
-		certs := x509.NewCertPool()
-		if !certs.AppendCertsFromPEM(ct.rootCerts) {
-			return errors.New("Unable to load root certificates")
+		config := ct.tlsConfig
+		if config == nil {
```

Comment:
My intention was that if a `nil` is passed through `NewTLSConnectionWithTLSConfig`, then `config = ct.tlsConfig` will be `nil`, and we'll get the `ct.rootCerts` below.  I think that's actually happening, I don't see a bug. Is it possible you misread the above?

Happy to change this code around if it's confusing, but I don't see what it's not correct ATM.  Thanks!